auth_mellon_util.c

```diff
@@ -1852,7 +2285,7 @@ bool am_is_paos_request(request_rec *r)
         }
     }
     if (paos_header) {
-        if (am_validate_paos_header(r, paos_header)) {
+        if (am_parse_paos_header(r, paos_header, &ecp_service_options)) {
             valid_paos_header = true;
```

Comment:
Yes, returning bad request would be the right behavior. Added error code return to am_parse_paos_header() and check it.